src/controllers/cohort.js

```diff
@@ -1,14 +1,24 @@
 import { createCohort } from '../domain/cohort.js'
 import { sendDataResponse, sendMessageResponse } from '../utils/responses.js'
+import * as validation from '../utils/validationFunctions.js'
 import ERR from '../utils/errors.js'
 
 export const create = async (req, res) => {
+  const { startDate, endDate } = req.body
+
   try {
-    const createdCohort = await createCohort()
+    const { parsedStartDate, parsedEndDate } = validation.dateValidation(
+      startDate,
+      endDate
+    )
+    const createdCohort = await createCohort(parsedStartDate, parsedEndDate)
 
     return sendDataResponse(res, 201, createdCohort)
   } catch (e) {
-    console.log('An error occured while trying to create a cohort:', e)
-    return sendMessageResponse(res, 500, { error: ERR.INTERNAL_ERROR })
+    console.error('error creating cohort:', e)
+    if (e.message === ERR.DATE_REQUIRED || e.message === ERR.DATE_FORMATTING) {
```

Comment:
It would be better to see if these fields are provided prior to making a database request